frontend/www/js/omegaup/components/problem/CollectionList.vue

```diff
@@ -31,7 +35,7 @@
           :path="`/problem/collection/${level}/`"
           @apply-filter="
             (columnName, sortOrder) =>
```

Comment:
Esto es por lo que no te funciona el filtro, falta que recibas el parámetro de dificultad, para que lo puedas emitir
```suggestion
            (columnName, sortOrder, difficulty) =>
```